SofaKernel/modules/SofaDeformable/PolynomialSpringsForceField.inl

```diff
@@ -0,0 +1,598 @@
+    }
+}
+
+
+template<class DataTypes>
+void PolynomialSpringsForceField<DataTypes>::draw(const core::visual::VisualParams* vparams)
+{
+    if (!((this->mstate1 == this->mstate2)?vparams->displayFlags().getShowForceFields():vparams->displayFlags().getShowInteractionForceFields()))
+        return;
+
+    vparams->drawTool()->saveLastState();
+
+    const VecCoord& p1 =this->mstate1->read(core::ConstVecCoordId::position())->getValue();
+    const VecCoord& p2 =this->mstate2->read(core::ConstVecCoordId::position())->getValue();
+
+    const VecIndex& firstObjectIndices = d_firstObjectPoints.getValue();
+    const VecIndex& secondObjectIndices = d_secondObjectPoints.getValue();
+
+    std::vector< defaulttype::Vector3 > points;
+    for (unsigned int i = 0; i < firstObjectIndices.size(); i++)
+    {
+        const unsigned int index1 = firstObjectIndices[i];
+        points.push_back(p1[index1]);
+        const unsigned int index2 = secondObjectIndices[i];
+        points.push_back(p2[index2]);
+    }
+
+    if (d_showArrowSize.getValue() == 0 || d_drawMode.getValue() == 0)
+    {
+        vparams->drawTool()->drawLines(points, 1, d_springColor.getValue());
+    }
+    else if (d_drawMode.getValue() == 1)
+    {
+        const unsigned int numLines = points.size() / 2;
+        for (unsigned int i = 0; i < numLines; ++i) {
+            vparams->drawTool()->drawCylinder(points[2*i+1], points[2*i], d_showArrowSize.getValue(), d_springColor.getValue());
+        }
+    }
+    else if (d_drawMode.getValue() == 2)
+    {
+        const unsigned int numLines = points.size() / 2;
+        for (unsigned int i = 0; i < numLines; ++i) {
+            vparams->drawTool()->drawArrow(points[2*i+1], points[2*i], d_showArrowSize.getValue(), d_springColor.getValue());
+        }
+    }
+
+    // draw connected point indices
+    defaulttype::Vec4f color(1.0, 1.0, 1.0, 1.0);
+
+    Real scale = (vparams->sceneBBox().maxBBox() - vparams->sceneBBox().minBBox()).norm() * d_showIndicesScale.getValue();
+
+    helper::vector<defaulttype::Vector3> positions;
+    for (size_t i = 0; i < firstObjectIndices.size(); i++) {
+        const unsigned int index = firstObjectIndices[i];
+        positions.push_back(defaulttype::Vector3(p1[index][0], p1[index][1], p1[index][2] ));
+    }
+    for (size_t i = 0; i < secondObjectIndices.size(); i++) {
+        const unsigned int index = secondObjectIndices[i];
+        positions.push_back(defaulttype::Vector3(p2[index][0], p2[index][1], p2[index][2] ));
+    }
+
+    vparams->drawTool()->drawPoints(positions, scale, color);
+
+    vparams->drawTool()->restoreLastState();
+}
+
+
+
+template<class DataTypes>
+void PolynomialSpringsForceField<DataTypes>::addKToMatrix(const core::MechanicalParams* mparams,
+                                                          const sofa::core::behavior::MultiMatrixAccessor* matrix )
+{
+    msg_info() << "[" <<  this->getName() << "]: addKToMatrix";
```

Comment:
why doing manually the [this->getName()] ? 
printing addKToMatrix seems to be a debugging feature... and is not supposed to be a real message for the user. do use dmsg_info() .